stacks-core/src/utils.rs

```diff
@@ -111,6 +111,30 @@ impl From<StacksAddress> for PrincipalData {
 	}
 }
 
+impl From<String> for PrincipalData {
+	fn from(string: String) -> Self {
+		let parts: Vec<&str> = string.split('.').collect();
+
+		match parts.len() {
+			1 => {
+				let address = StacksAddress::try_from(parts[0]).unwrap();
+
+				Self::Standard(StandardPrincipalData::from(address))
+			}
+			2 => {
+				let address = StacksAddress::try_from(parts[0]).unwrap();
+				let contract_name = ContractName::new(parts[1]).unwrap();
```

Comment:
Given that you're unwrapping in this `From<String>` implementation this would be much better as a `TryFrom<String>` implementation.